amundsen_application/static/js/pages/DashboardPage/index.tsx

```diff
@@ -121,52 +129,54 @@ export class DashboardPage extends React.Component<
 
   renderTabs() {
     const tabInfo: TabInfo[] = [];
+    const { dashboard } = this.props;
 
     tabInfo.push({
       content: (
         <ResourceList
-          allItems={this.props.dashboard.tables}
+          allItems={dashboard.tables}
           itemsPerPage={TABLES_PER_PAGE}
           source={DASHBOARD_SOURCE}
         />
       ),
       key: 'tables',
-      title: `Tables (${this.props.dashboard.tables.length})`,
+      title: `Tables (${dashboard.tables.length})`,
     });
 
-    if (this.props.dashboard.chart_names.length > 0) {
+    if (dashboard.chart_names.length > 0) {
       tabInfo.push({
-        content: <ChartList charts={this.props.dashboard.chart_names} />,
+        content: <ChartList charts={dashboard.chart_names} />,
         key: 'charts',
-        title: `Charts (${this.props.dashboard.chart_names.length})`,
+        title: `Charts (${dashboard.chart_names.length})`,
       });
     }
 
     tabInfo.push({
       content: (
-        <QueryList
-          product={this.props.dashboard.product}
-          queries={this.props.dashboard.queries}
-        />
+        <QueryList product={dashboard.product} queries={dashboard.queries} />
       ),
       key: 'queries',
-      title: `Queries (${this.props.dashboard.queries.length})`,
+      title: `Queries (${dashboard.queries.length})`,
     });
 
     return <TabsComponent tabs={tabInfo} defaultTab="tables" />;
   }
 
   render() {
-    const { dashboard, isLoading } = this.props;
+    const { dashboard, isLoading, statusCode } = this.props;
     const hasDescription =
       dashboard.description && dashboard.description.length > 0;
     const hasLastRunState =
       dashboard.last_run_state && dashboard.last_run_state.length > 0;
+    const dashboardNotice = getResourceNotices(
+      ResourceType.dashboard,
+      `${dashboard.product}.${dashboard.name}`
```

Comment:
I don't think the product and dashboard name are enough to uniquely identify a dashboard. Multiple dashboards in the same product are named some generic "payments dashboard" or something that would all match the same config value. The "uri" ("key" for tables, we should standardize to one or the other but that's a different issue) for dashboards is `{product}://{cluster}.{group}/{name}`.

Reply:
Good call, I will try to add more stuff here.